### PR TITLE
TIG-1404 bubble up test failures

### PR DESCRIPTION
**Pull request by guoyr**
As part of this work
- fixed global rate limiter test
- split up the lamp script
- added unit test for the new test failure code.

### CODE REVIEW

File: scripts/lamplib/tasks/run_tests.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import os
```
Comment (guoyr) [inline scripts/lamplib/tasks/run_tests.py]: This file and `test_run_tests.py` are the only two pieces of new Python code. The rest of the Python files is mostly splitting up the original script.

File: scripts/lamplib/__main__.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        tasks.compile_all(env)
+
+    if args.subcommand == 'install':
+        tasks.install(env)
+    elif args.subcommand == 'cmake-test':
+        tasks.run_tests.cmake_test(env)
+
+
+if __name__ == '__main__':
+    main()
```
Comment (rtimmons) [inline scripts/lamplib/__main__.py]: I like moving/splitting the files but can we please put inside `src/python`?

File: scripts/lamplib/context.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class Context:
+    # Permanent constants.
+    TOOLCHAIN_BUILD_ID = '7297534c3dc1df9f6d315098174171eaf75c5846_19_02_09_02_19_44'
+    TOOLCHAIN_GIT_HASH = TOOLCHAIN_BUILD_ID.split('_')[0]
+
+    # User configurable constants.
+    IGNORE_TOOLCHAIN_VERSION = False
+
+    # User configurable dynamic context.
+    get_compile_environment = _create_compile_environment
```
Comment (rtimmons) [inline scripts/lamplib/context.py]: why not define it in the class?

```python
class Context:
  def get_compile_enviornment(self, ...): ...
```

File: evergreen.yml [hunk 4]
```diff
@@ -418,6 +419,25 @@ functions:
...
+      shell: bash
+      script: |
+        set -o errexit
+        set -o pipefail
+        set -o nounset
+
+        # for python3
+        export PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+
+        ./scripts/lamp self-test
```
Comment (rtimmons) [inline evergreen.yml]: this new task goes away if you put `lamplib` into `src/python`, ya?

**Pull request merged**